terraform/environment/load_balancer_public_front.tf

```
@@ -48,6 +48,52 @@ resource "aws_lb_listener" "public_front_loadbalancer" {
   }
 }
 
+resource "aws_ssm_parameter" "maintenance_switch" {
+  name            = "${local.environment}_enable_maintenance"
+  type            = "String"
+  value           = "false"
+  description     = "values of either 'true' or 'false' only"
+  allowed_pattern = "^(true|false)"
+  overwrite       = true
+  lifecycle {
+    ignore_changes = [
+      "value",
+    ]
+  }
+}
+
+
+locals {
+  path_pattern = {
+    field  = "path-pattern"
+    values = ["/maintenance"]
+  }
+  host_pattern = {
+    field  = "host-header"
+    values = [local.public_front_fqdn_to_bypass_in_maintenance]
+  }
+  rule_condition                             = aws_ssm_parameter.maintenance_switch.value == "true" ? local.host_pattern : local.path_pattern
+  public_front_fqdn_to_bypass_in_maintenance = local.environment == "production" ? aws_route53_record.public_front.fqdn : aws_route53_record.public_front.fqdn
```

magic string check here, I'd prefer to see this as environment specific setting in your config. that said, the conditional is just a noop right now so you may as well have....
```suggestion
  public_front_fqdn_to_bypass_in_maintenance =  aws_route53_record.public_front.fqdn
```
did you mean to do something else here?